fiona/_shim1.pyx

```diff
@@ -50,7 +50,10 @@ cdef void* gdal_open_vector(const char *path_c, int mode, drivers, options):
                 #collection._driver = name
                 break
     else:
-        cogr_ds = OGROpen(path_c, mode, NULL)
+        try:
+            cogr_ds = exc_wrap_pointer(OGROpen(path_c, mode, NULL))
+        except Exception as exc:
```

Comment:
I think we should catch only `CPLE_BaseError` here so we don't accidentally hide any errors related to malformed paths or bogus modes.